## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 512148`
Commit: [`3c3b7c8a77bd83b0fd83c2d747674e70c37e5f8e`](https://github.com/NixOS/nixpkgs/commit/3c3b7c8a77bd83b0fd83c2d747674e70c37e5f8e) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/3c3b7c8a77bd83b0fd83c2d747674e70c37e5f8e..pull/512148/head))
Merge: [`44ea805241f23ec2cf38e05a79ca788178a3e86f`](https://github.com/NixOS/nixpkgs/commit/44ea805241f23ec2cf38e05a79ca788178a3e86f)

Logs: https://github.com/limwa/nixpkgs-review-gha/actions/runs/25145413824


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 30 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.virtualbox</li>
    <li>linuxKernel.packages.linux_5_10.virtualboxGuestAdditions</li>
    <li>linuxKernel.packages.linux_5_15.virtualbox</li>
    <li>linuxKernel.packages.linux_5_15.virtualboxGuestAdditions</li>
    <li>linuxKernel.packages.linux_6_1.virtualbox</li>
    <li>linuxKernel.packages.linux_6_1.virtualboxGuestAdditions</li>
    <li>linuxKernel.packages.linux_6_12.virtualbox</li>
    <li>linuxKernel.packages.linux_6_12.virtualboxGuestAdditions</li>
    <li>linuxPackages.virtualbox (linuxKernel.packages.linux_6_18.virtualbox)</li>
    <li>linuxPackages.virtualboxGuestAdditions (linuxKernel.packages.linux_6_18.virtualboxGuestAdditions)</li>
    <li>linuxKernel.packages.linux_6_6.virtualbox</li>
    <li>linuxKernel.packages.linux_6_6.virtualboxGuestAdditions</li>
    <li>linuxPackages_latest.virtualbox (linuxKernel.packages.linux_7_0.virtualbox)</li>
    <li>linuxPackages_latest.virtualboxGuestAdditions (linuxKernel.packages.linux_7_0.virtualboxGuestAdditions)</li>
    <li>linuxPackages_xanmod.virtualbox (linuxKernel.packages.linux_xanmod.virtualbox)</li>
    <li>linuxPackages_xanmod.virtualboxGuestAdditions (linuxKernel.packages.linux_xanmod.virtualboxGuestAdditions)</li>
    <li>linuxPackages_xanmod_latest.virtualbox (linuxKernel.packages.linux_xanmod_latest.virtualbox, linuxPackages_xanmod_stable.virtualbox)</li>
    <li>linuxPackages_xanmod_latest.virtualboxGuestAdditions (linuxKernel.packages.linux_xanmod_latest.virtualboxGuestAdditions, linuxPackages_xanmod_stable.virtualboxGuestAdditions)</li>
    <li>linuxPackages_zen.virtualbox (linuxKernel.packages.linux_zen.virtualbox)</li>
    <li>linuxPackages_zen.virtualboxGuestAdditions (linuxKernel.packages.linux_zen.virtualboxGuestAdditions)</li>
    <li>virtualbox</li>
    <li>virtualbox.modsrc</li>
    <li>virtualboxExtpack</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHardened.modsrc</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxHeadless.modsrc</li>
    <li>virtualboxKvm</li>
    <li>virtualboxWithExtpack</li>
    <li>virtualboxWithExtpack.modsrc</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.virtualboxGuestAdditions</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.virtualboxGuestAdditions</li>
    <li>linuxKernel.packages.linux_6_1.virtualboxGuestAdditions</li>
    <li>linuxKernel.packages.linux_6_12.virtualboxGuestAdditions</li>
    <li>linuxPackages.virtualboxGuestAdditions (linuxKernel.packages.linux_6_18.virtualboxGuestAdditions)</li>
    <li>linuxKernel.packages.linux_6_6.virtualboxGuestAdditions</li>
    <li>linuxPackages_latest.virtualboxGuestAdditions (linuxKernel.packages.linux_7_0.virtualboxGuestAdditions)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>linuxKernel.packages.linux_5_10.virtualboxGuestAdditions</summary>
<pre>=== Building &#x27;vboxsf&#x27; module ===
make[1]: Entering directory &#x27;/build/src/vboxguest-7.2.8_NixOS/vboxsf&#x27;
make V= CONFIG_MODULE_SIG= CONFIG_MODULE_SIG_ALL= -C /nix/store/y6f6cn9714wr4za63j1nv3znia03my83-linux-5.10.253-dev/lib/modules/5.10.253/build M=/build/src/vboxguest-7.2.8_NixOS/vboxsf SRCROOT=/build/src/vboxguest-7.2.8_NixOS/vboxsf -j4 modules
  CC [M]  /build/src/vboxguest-7.2.8_NixOS/vboxsf/vfsmod.o
  CC [M]  /build/src/vboxguest-7.2.8_NixOS/vboxsf/dirops.o
  CC [M]  /build/src/vboxguest-7.2.8_NixOS/vboxsf/lnkops.o
  CC [M]  /build/src/vboxguest-7.2.8_NixOS/vboxsf/regops.o
/build/src/vboxguest-7.2.8_NixOS/vboxsf/regops.c: In function &#x27;vbsf_lock_user_pages_failed_check_kernel&#x27;:
/build/src/vboxguest-7.2.8_NixOS/vboxsf/regops.c:1452:24: error: &#x27;TASK_SIZE_MAX&#x27; undeclared (first use in this function); did you mean &#x27;TASK_SIZE_32&#x27;?
 1452 |         &amp;&amp; uPtrFrom &gt;= TASK_SIZE_MAX)
      |                        ^~~~~~~~~~~~~
      |                        TASK_SIZE_32
/build/src/vboxguest-7.2.8_NixOS/vboxsf/regops.c:1452:24: note: each undeclared identifier is reported only once for each function it appears in
make[4]: *** [/nix/store/y6f6cn9714wr4za63j1nv3znia03my83-linux-5.10.253-dev/lib/modules/5.10.253/source/scripts/Makefile.build:286: /build/src/vboxguest-7.2.8_NixOS/vboxsf/regops.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [/nix/store/y6f6cn9714wr4za63j1nv3znia03my83-linux-5.10.253-dev/lib/modules/5.10.253/source/Makefile:1853: /build/src/vboxguest-7.2.8_NixOS/vboxsf] Error 2
make[2]: *** [/nix/store/y6f6cn9714wr4za63j1nv3znia03my83-linux-5.10.253-dev/lib/modules/5.10.253/source/Makefile:192: __sub-make] Error 2
make[1]: *** [/build/src/vboxguest-7.2.8_NixOS/vboxsf/Makefile-footer.gmk:146: vboxsf] Error 2
make[1]: Leaving directory &#x27;/build/src/vboxguest-7.2.8_NixOS/vboxsf&#x27;
make: *** [Makefile:60: vboxsf] Error 1</pre>
</details>
</details>

--- a/nixos/modules/hardware/video/virtualbox.nix
+++ b/nixos/modules/hardware/video/virtualbox.nix
@@ -4,7 +4,18 @@ let
   inherit (config.services.xserver) videoDrivers;
 in
 {
-  boot.extraModulePackages = lib.mkIf (lib.elem "virtualbox" videoDrivers) [
-    kernelPackages.virtualboxGuestAdditions
-  ];
+  config = lib.mkIf (lib.elem "virtualbox" videoDrivers) {
+    assertions = [
+      {
+        assertion = lib.versionOlder kernelPackages.kernel.version "7.0";
+        message = ''
+          The `virtualbox` video driver provided by VirtualBox Guest Additions has been deprecated upstream for Linux kernel 7.0+.
+        '';
+      }
+    ];
+
+    boot.extraModulePackages = [
+      kernelPackages.virtualboxGuestAdditions
+    ];
+  };
 }

--- a/nixos/tests/virtualbox.nix
+++ b/nixos/tests/virtualbox.nix
@@ -505,12 +505,12 @@ mapAttrs (mkVBoxTest { } vboxVMs) {
   simple-gui = ''
     # Home to select Tools, down to move to the VM, enter to start it.
     def send_vm_startup():
-        machine.send_key("home")
-        machine.send_key("down")
+        machine.send_key("ctrl-m")
         machine.send_key("ret")
 
 
     create_vm_simple()
+    machine.succeed(ru("VBoxManage setextradata global \"GUI/Input/SelectorShortcuts\" \"ToolsGlobalMachineManager=Ctrl+M\""))
     machine.succeed(ru("VirtualBox >&2 &"))
     machine.wait_until_succeeds(ru("xprop -name 'Oracle VirtualBox Manager'"))
     machine.sleep(5)

--- a/nixos/tests/virtualbox.nix
+++ b/nixos/tests/virtualbox.nix
@@ -77,6 +77,8 @@ let
 
       virtualisation.virtualbox.guest.enable = true;
 
+      boot.initrd.systemd.enable = false;
+
       boot.initrd.kernelModules = [
         "af_packet"
         "vboxsf"

